src/model.js

```diff
@@ -2544,6 +2544,10 @@ Specify a different name for either index to resolve this issue.`);
     const updateValues = Utils.mapValueFieldNames(updatedDataValues, options.fields, this);
     const now = Utils.now(this.sequelize.dialect);
 
+    options.conflictFields = options.conflictFields?.map(
+      attr => (this.rawAttributes[attr]?.field) || attr,
```

Comment:
Since this is a breaking change, should we rename `conflictFields` to `conflictAttributes` in this PR? (see comment in https://github.com/sequelize/sequelize/pull/13419)